### DISCUSSION TITLE
[CLI/PNG] Exporting to PNG always fills the background with white color

### DISCUSSION THREAD

**Discussion by dahooz** [typst/typst#3914] [General]
When using the CLI to export to PNG, the background is always filled with `Color::WHITE`:

https://github.com/typst/typst/blob/72dd79210602ecc799726fc096b078afbb47f299/crates/typst-cli/src/compile.rs#L279

I'd expect the background to be filled with the `#page(fill: "...")` color. Is there a reason for the white background, or does it make sense to open a PR?

**Comment by LaurenzV** [comment]
The reason is probably because PDFs by default are also white. However, if you do something like `#set page(fill: red)` then the page should also be red in the PNG export. It will basically draw red on top of the white background.

**Reply by dahooz** [reply]
I'm pretty sure that the white background in PDFs is rendering, not an actual background:
Due to this hardcoded color, I currently use imagemagick's `convert` to convert from PDF to PNG. This leads to a transparent background in the png output.

Maybe I should make my usecase more clear: I'm creating "heading" PNGs with fancy `#text` settings, that I can use in other documents (read: usual office software).

**Reply by LaurenzV** [reply]
I mean I don't see why this shouldn't be configurable. We could add some kind of CLI argument that sets the background color for PNG outputs.

**Comment by laurmaedje** [comment]
In addition to what @LaurenzV said, the `page.fill` property can also take various other kinds of fills (patterns, gradients), so we couldn't fully handle it there anyway.

**Comment by dahooz** [comment]
With @laurmaedje 's argument in mind, I would argue that it doesn't actually make sense to allow inputting background colors via CLI argument: I can imagine someone expecting to be able to put sth. like `rgb("#abcdef").darken(80%).desaturate(20%)...` in that flag, requiring to parse typst code from this flag...

I'd rather just add a `--png-fill-transparent` flag, that changes the `Color::WHITE` to `#rgb("#00000000")` (sorry, didn't find how to write the equivalent in Rust). Then it's just the `page.fill` property handling the background.
